src/borg/archiver/extract_cmd.py

```diff
@@ -71,29 +75,34 @@ def do_extract(self, args, repository, manifest, archive):
                     except BackupError as e:
                         self.print_warning_instance(BackupWarning(remove_surrogates(dir_item.path), e))
 
-                if output_list and matcher.match(item.path):
-                    logging.getLogger("borg.output.list").info(remove_surrogates(item.path))
+            is_matched = matcher.match(item.path)
             try:
+                log_prefix = "+" if is_matched else "-"
+                logging.getLogger("borg.output.list").info(f"{log_prefix} {remove_surrogates(item.path)}")
                 if dry_run:
-                    if matcher.match(item.path):
-                        logging.getLogger("borg.output.list").info(f"+ {remove_surrogates(item.path)}")
-                    else:
-                        logging.getLogger("borg.output.list").info(f"- {remove_surrogates(item.path)}")
                     archive.extract_item(item, dry_run=True, hlm=hlm, pi=pi)
                 else:
-                    if matcher.match(item.path):
+                    if is_matched:
                         if stat.S_ISDIR(item.mode):
                             dirs.append(item)
                             archive.extract_item(item, stdout=stdout, restore_attrs=False)
-                    else:
-                        archive.extract_item(
-                            item, stdout=stdout, sparse=sparse, hlm=hlm, pi=pi, continue_extraction=continue_extraction
-                        )
+                        else:
+                            archive.extract_item(
+                                item,
+                                stdout=stdout,
+                                sparse=sparse,
+                                hlm=hlm,
+                                pi=pi,
+                                continue_extraction=continue_extraction,
+                            )
             except BackupError as e:
                 self.print_warning_instance(BackupWarning(remove_surrogates(orig_path), e))
-
         if pi:
             pi.finish()
+        # Display excluded items
+        if excluded_items:
+            for excluded_item in excluded_items:
+                logging.getLogger("borg.output.list").info(f"- {remove_surrogates(excluded_item)}")
```

Comment:
It would be better to display included and excluded items both together in the order they appear in the archive.

Reply:
@ThomasWaldmann done.